src/goto-symex/symex_assign.cpp

```diff
@@ -67,6 +67,31 @@ expr_skeletont expr_skeletont::compose(expr_skeletont other) const
   return expr_skeletont(apply(other.skeleton));
 }
 
+optionalt<expr_skeletont> expr_skeletont::array() const
+{
+  if(skeleton.id() != ID_index)
+    return {};
+  return expr_skeletont{to_index_expr(skeleton).array()};
+}
+
+optionalt<expr_skeletont> expr_skeletont::compound() const
+{
+  if(skeleton.id() != ID_member)
+    return {};
+  return expr_skeletont{to_member_expr(skeleton).compound()};
+}
+
+optionalt<expr_skeletont> expr_skeletont::object() const
```

Comment:
`byte_extract_or_member_object`? Again I'd expect `object` to address `index_exprt` too.

Reply:
Actually the `member` part was a mistake, I will make it `byte_extract_object`